src/prefect/orion/schemas/schedules.py

```diff
@@ -359,32 +359,65 @@ def validate_rrule_str(cls, v):
 
     @classmethod
     def from_rrule(cls, rrule: dateutil.rrule.rrule):
-        if not isinstance(rrule, dateutil.rrule.rrule):
-            raise ValueError(f"Invalid RRule object: {rrule}")
-        if rrule._dtstart.tzinfo is not None:
-            timezone = rrule._dtstart.tzinfo.name
+        if isinstance(rrule, dateutil.rrule.rrule):
+            if rrule._dtstart.tzinfo is not None:
+                timezone = rrule._dtstart.tzinfo.name
+            else:
+                timezone = "UTC"
+            return RRuleSchedule(rrule=str(rrule), timezone=timezone)
+        elif isinstance(rrule, dateutil.rrule.rruleset):
+            dtstarts = [rr._dtstart for rr in rrule._rrule if rr._dtstart is not None]
+            unique_dstarts = set(pendulum.instance(d).in_tz("UTC") for d in dtstarts)
+            unique_timezones = set(d.tzinfo for d in dtstarts if d.tzinfo is not None)
+
+            if len(unique_timezones) > 1:
+                raise ValueError(f"rruleset has too many dtstart timezones: {rrule}")
```

Comment:
Should we also include which timezones we found?

Something like:
```suggestion
                raise ValueError(f"rruleset {rrule} has too many dtstart timezones: {unique_timezones}")
```

I'm not sure what the `repr(rrule)` here will look like, maybe it has all the info we'd need?

Reply:
it doesn't, specifying the timezones is a good idea